src/booking/PartialOfferRequests/PartialOfferRequests.ts

```diff
@@ -0,0 +1,74 @@
+import { Client } from 'Client'
+import { Resource } from '../../Resource'
+import { CreateOfferRequest, DuffelResponse, OfferRequest } from '../../types'
+import { SelectedPartialOffersParams } from './PartialOfferRequestTypes'
+
+/**
+ * To search for and select flights separately for each slice of the journey, you'll need to create a partial offer request.
+ *
+ * A partial offer request describes the passengers and where and when they want to travel (in the form of a list of slices).
+ * @class
+ * @link https://duffel.com/docs/api/partial-offer-requests
+ */
+export class PartialOfferRequests extends Resource {
+  /**
+   * Endpoint path
+   */
+  path: string
+
+  constructor(client: Client) {
+    super(client)
+    this.path = 'air/partial_offer_requests'
+  }
+
+  /**
+   * Retrieves a partial offers request by its ID, only including partial offers for the current slice of multi-step search flow.
+   * @param {string} id - Duffel's unique identifier for the offer request
+   * @param {Object} [options] - Selected partial offers
+   * @link https:/duffel.com/docs/api/partial-offer-requests/get-partial-offer-request-by-id
+   */
+  public get = async (
+    id: string,
+    options?: SelectedPartialOffersParams
+  ): Promise<DuffelResponse<OfferRequest>> => {
+    return this.request({
+      method: 'GET',
+      path: `${this.path}/${id}`,
+      params: options,
+    })
+  }
+
+  /**
+   * To search for and select flights separately for each slice of the journey, you'll need to create a partial offer request.
+   * A partial offer request describes the passengers and where and when they want to travel (in the form of a list of slices).
+   * It may also include additional filters (e.g. a particular cabin to travel in).
+   * @param {Object} [options] - the parameters for making a partial offer requests (required: slices, passengers; optional: cabin_class)
+   * @link https://duffel.com/docs/api/partial-offer-requests/create-partial-offer-request
+   */
+  public create = async <QueryParams>(
+    options: CreateOfferRequest & QueryParams
+  ): Promise<DuffelResponse<OfferRequest>> => {
+    return this.request({
+      method: 'POST',
+      path: `${this.path}/`,
+      data: options,
+    })
+  }
+
+  /**
+   * Retrieves an offer request with offers for fares matching selected partial offers..
+   * @param {string} id - Duffel's unique identifier for the offer request
+   * @param {Object} [options] - Selected partial offers
+   * @link https:/duffel.com/docs/api/partial-offer-requests/get-partial-offer-request-fares-by-id
+   */
+  public fares = async (
```

Comment:
Since the idea it's to get fares by id could you update the function to be `getFaresById`? I think it makes more sense.

Reply:
I could, but that would make it even more complicated to generate the examples, as `getFaresById` is not in the definition anywhere

Reply:
Updated, will figure out how to deal with example docs separately.

Reply:
We can roll back to `fares` if it's really complicated.